feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/networkManagement/networkList/common/NetworkListAdapterItemFactory.kt

```diff
@@ -0,0 +1,56 @@
+package io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common
+
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.utils.images.asIcon
+import io.novafoundation.nova.feature_settings_impl.R
+import io.novafoundation.nova.feature_settings_impl.domain.NetworkState
+import io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter.items.NetworkListNetworkRvItem
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novasama.substrate_sdk_android.wsrpc.state.SocketStateMachine
+
+interface NetworkListAdapterItemFactory {
+
+    fun getNetworkItem(network: NetworkState): NetworkListNetworkRvItem
+}
+
+class RealNetworkListAdapterItemFactory(
+    private val resourceManager: ResourceManager
+) : NetworkListAdapterItemFactory {
+
+    override fun getNetworkItem(network: NetworkState): NetworkListNetworkRvItem {
+        val chain = network.chain
+        val chainEnabled = true // TODO: Update it after modifying chain db model
```

Comment:
Can already use `chain.connectionState != Chain.ConnectionState.DISABLED`, maybe create it as `Chain.isEnabled` extension